### PR TITLE
Fix/ Abstract which contains Mathjax formulas

### PR DESCRIPTION
**Pull request by xkopenreview**
Some abstract may contain mathjax formulas which will break the abstract display in forum page
an example is :
```
pacity of the DTPC with ISI when the size of the target message set 
<inline-formula xmlns:mml=&quot;http://www.w3.org/1998/Math/MathML&quot; xmlns:xlink=&quot;http://www.w3.org/1999/xlink&quot;> <tex-math notation=&quot;LaTeX&quot;>$K$ </tex-math></inline-formula>
 and the number of ISI channel taps 
``` 

this pr should remove the inline-formula tags and keep only 
```
pacity of the DTPC with ISI when the size of the target message set $K$ and the number of ISI channel taps 
```  
which will be rendered as 
>pacity of the DTPC with ISI when the size of the target message set $K$ and the number of ISI channel taps

 correctly by forum page

this pr should also update redirection logic

### CODE REVIEW

File: packages/meta-extraction/src/abstractExtractionRules.js [hunk 1]
```diff
@@ -342,6 +342,22 @@ const cleanAbstract = (abstract) => {
...
+      tempDiv.innerHTML = p;
+      // tempDiv.style['white-pace'] = 'pre-line';
+      const extractedText = tempDiv.textContent.trim();
+      return extractedText;
+  }, str);
+  } catch (error) {
+    return str;
+  }
+};
+
```
[the remark this answers is not in the document]
Comment (xkopenreview) [reply packages/meta-extraction/src/abstractExtractionRules.js:360]: if this abstract is captured by the regex then the tags will be removed
the style will be lost though

[the remark this answers is not in the document]
Comment (carlosmondra) [reply packages/meta-extraction/src/abstractExtractionRules.js]: I see. I was wondering if we should do this, but maybe it's a different discussion. I'm not sure what to do about that. The issue is present in the method that converts dblp xml to a note.

File: packages/meta-extraction/src/index.js [hunk 2]
```diff
@@ -52,11 +52,17 @@ const extractAbstract = async (url, skipTidy = false) => {
...
     const status = response.status().toString();
+    const responseUrl = response.url();
     console.log(`HTTP status code: ${status}`);
 
-    if (shouldEnableMultiRedirect(response.url())){
+    if (shouldEnableMultiRedirect(responseUrl)){
       await browserInstance.close();
-      return extractAbstract(response.url(), true);
+      if (status.startsWith('3') || status.startsWith('2')) {
+        console.log(`Redirecting to ${responseUrl}`);
```
Comment (carlosmondra) [inline packages/meta-extraction/src/index.js:61]: Do we need this console.log?

Comment (xkopenreview) [reply packages/meta-extraction/src/index.js:61]: all the console.log won't appear in process function log
they may be used when the tests are executed in github actions

File: packages/meta-extraction/src/abstractExtractionRules.js [hunk 3]
```diff
@@ -952,15 +973,16 @@ const runAllRules = async (html, page, url) => {
...
-      return {};
+    if (error) {
+      return {error};
     }
 
     if (abstract || pdf) {
+      let cleanedAbstract = await cleanMathjax(abstract,page);
       return {
-        abstract: cleanAbstract(abstract),
+        abstract: cleanAbstract(cleanedAbstract),
```
Comment (carlosmondra) [inline packages/meta-extraction/src/abstractExtractionRules.js:985]: Would it be better to call `cleanMathjax` inside `cleanAbstract`? To keep things cleaner (pun intended).

**Review by carlosmondra — APPROVED** [review]

**Pull request merged**